src/waveform/waveformwidgetfactory.cpp

```diff
@@ -613,7 +589,7 @@ void WaveformWidgetFactory::setPlayMarkerPosition(double position) {
 
 void WaveformWidgetFactory::notifyZoomChange(WWaveformViewer* viewer) {
     WaveformWidgetAbstract* pWaveformWidget = viewer->getWaveformWidget();
-    if (pWaveformWidget != NULL && isZoomSync()) {
+    if (pWaveformWidget != nullptr) {
```

Comment:
I think we should clean that a bit more and remove that case entirely. 
notifyZoomChange() should become setZoom(double zoom)
All zoom related calculation and virables should be moved to the WaveformWidgetFactor.

Reply:
I did a bit of refactoring in 6a15b74. Currently there is a `waveform_zoom`. This is awkward now that there is only one zoom level. Should we add a new `[Master], waveform_zoom` CO that changes all the legacy per-deck COs?

Reply:
@toszlanyi what do you think?

Reply:
> @toszlanyi what do you think?

In my opinion most users have waveforms in sync most of the time. I myself would love to keep them separated. Mostly I use one of the most zoom out up to 20% to keep overview about the tracks playing. Only when to set a CUE point or the like I zoom in. If you prepare a set then it doesnt matter but most of times even at a gig I dont and prepare while I am playing. So lets say I wanna keep overview about running tracks on Deck 1 and 2 and zoom in for channel 3 to set a hot cue for the next track. This happens not too often but it does. So in my experience I would love to keep the option.

Reply:
![grafik](https://user-images.githubusercontent.com/58949409/91841495-c6806c80-ec52-11ea-8d65-3740a5b5b1e0.png)

Reply:
> So lets say I wanna keep overview about running tracks on Deck 1 and 2 and zoom in for channel 3 to set a hot cue for the next track.

But why do you want to zoom the waveforms separately? It looks like it wouldn't really matter in your case.

Reply:
side note @toszlanyi The zoom buttons in the screenshot are lit, so I assume you use `engine.setValue(group, "waveform_zoom_up", 1)` without setting them back to 0.
Instead you could use `script.triggerControl(group, "waveform_zoom_up", T)` which sets the value back to 0 after T ms (100 recommended). see https://github.com/mixxxdj/mixxx/blob/master/res/controllers/common-controller-scripts.js#L208

Reply:
> > So lets say I wanna keep overview about running tracks on Deck 1 and 2 and zoom in for channel 3 to set a hot cue for the next track.
> 
> But why do you want to zoom the waveforms separately? It looks like it wouldn't really matter in your case.

Because then while I set or adjust a cue point on a track zoomed in I would not lose the overview of the actual playing track(s). Secondly maybe even more important is ... while I adjust details in one track that is stopped but the actual playing track or even 2 playing tracks are running with zoomed waveforms would really flash my eyes. This would be rather annoying to me. I strongly recommend to keep an option to separate zoom per deck even you may not see a personal benefit. Thx!

Reply:
> side note @toszlanyi The zoom buttons in the screenshot are lit, so I assume you use `engine.setValue(group, "waveform_zoom_up", 1)` without setting them back to 0.
> Instead you could use `script.triggerControl(group, "waveform_zoom_up", T)` which sets the value back to 0 after T ms (100 recommended). see https://github.com/mixxxdj/mixxx/blob/master/res/controllers/common-controller-scripts.js#L208

Wow! Thanks a lot for the hint. I gonna check this out. Cheers!

Reply:
If you're only zooming in for placing cue points, wouldn't you only be zoomed in briefly, then zoom out again? I still don't understand why you'd want the waveform zooms out of sync even if they aren't adjacent.

Reply:
> If you're only zooming in for placing cue points, wouldn't you only be zoomed in briefly, then zoom out again? I still don't understand why you'd want the waveform zooms out of sync even if they aren't adjacent.

That's absolutely right but still it interrupts the visual flow. I visualized that for you... 
https://www.dropbox.com/s/hyr4276ydwzhldt/zoom.mp4?dl=0

I need to admit that keep zoom in sync makes perfectly sense for parallel waveforms but not in the case that I use it. If you wanna delete the option in settings menu you may force zoom sync with parallel waveform and no sync when parallel waveform is switched off. 

Anyhow, thanks a lot for asking about my opinion, @Be-ing - much appreciate that.

Reply:
> If you wanna delete the option in settings menu you may force zoom sync with parallel waveform and no sync when parallel waveform is switched off.

That seems like a reasonable alternative.